src/app/clusters/icd-management-server/ICDManagementCluster.h

```diff
@@ -17,79 +17,86 @@
 
 #pragma once
 
-#include <app-common/zap-generated/cluster-objects.h>
-#include <app/AttributeAccessInterface.h>
-#include <app/CommandHandler.h>
 #include <app/icd/server/ICDConfigurationData.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <app/server-cluster/OptionalAttributeSet.h>
 #include <app/util/basic-types.h>
-#include <crypto/SessionKeystore.h>
+#include <clusters/IcdManagement/Commands.h>
+#include <clusters/IcdManagement/Metadata.h>
 #include <lib/core/Optional.h>
 #include <lib/support/Span.h>
-#include <protocols/interaction_model/StatusCode.h>
 
 #include <app/icd/server/ICDServerConfig.h>
 
 #if CHIP_CONFIG_ENABLE_ICD_CIP
-#include <app/ConcreteAttributePath.h>
-#include <app/icd/server/ICDMonitoringTable.h> // nogncheck
 #include <credentials/FabricTable.h>
+#include <crypto/SessionKeystore.h>
 #include <lib/core/CHIPPersistentStorageDelegate.h>
 #endif // CHIP_CONFIG_ENABLE_ICD_CIP
 
-namespace chip {
-namespace Crypto {
-using SymmetricKeystore = SessionKeystore;
-} // namespace Crypto
-} // namespace chip
-
 namespace chip {
 namespace app {
 namespace Clusters {
 
-class ICDManagementServer
+namespace IcdManagement {
+enum class OptionalCommands : uint32_t
+{
+    kStayActive = 0x01,
+};
+
+constexpr size_t kUserActiveModeTriggerInstructionMaxLength = 128;
+} // namespace IcdManagement
+
+/**
+ * @brief ICD Management Cluster
+ *
+ * This class provides the core ICD Management functionality. When CIP (Check-In Protocol)
+ * features are needed, use ICDManagementClusterWithCIP which extends this class.
+ */
+class ICDManagementCluster : public DefaultServerCluster
 {
 public:
-    ICDManagementServer() = default;
+    using OptionalAttributeSet = chip::app::OptionalAttributeSet<IcdManagement::Attributes::UserActiveModeTriggerInstruction::Id>;
```

Comment:
```suggestion
    using OptionalAttributeSet = app::OptionalAttributeSet<IcdManagement::Attributes::UserActiveModeTriggerInstruction::Id>;
```
In general, there are lots of extra prefixes various places here that should be removed.